core/contracts/composed_stream_template.kf

```diff
@@ -779,7 +779,7 @@ procedure get_current_version($show_disabled bool) private view returns (result
 // - gets an array of values for each properties, zipping them as needed
 // - increases the version
 // - adds taxonomy records in batch
-procedure set_taxonomy($data_providers text[], $stream_ids text[], $weights decimal(36,18)[]) public {
+procedure set_taxonomy($data_providers text[], $stream_ids text[], $weights decimal(36,18)[], $start_dates text[]) public {
     stream_owner_only();
```

Comment:
maybe there could be just one start date. I mean, the intention is that a whole set of taxonomies is valid from a point forward.

Each child having a different start date on the same set of taxonomy could be very complex, I think, unless there's a good reason.